### DISCUSSION TITLE
Usi API Manifests with the Canvas Learning Management System

### DISCUSSION THREAD

**Discussion by scotwoodyard-csulb** [microsoft/semantic-kernel#8372] [General]
I am interested in integrating Semantic Kernel with the Canvas Learning Management System using the Canvas API.  Kiota does not list the Canvas API, but I do have the Swagger 2.0 definition here:
https://github.com/buckett/canvas-api-client-feign/blob/master/canvas-swagger2.json

Is this possible?

**Comment by moonbox3** [comment]
Hi @scotwoodyard-csulb, are you looking to do this in SK dotnet, Python, or Java? 

Each language supports adding a plugin via an OpenAPI spec, but if working in Python, the spec would need to be upgraded to be 3.X+ instead of its current version of 2.0. I believe dotnet can handle both 2.0 and 3.0.
